This patch fixes nfs-utils to support nested builds in subdirectories
properly.

 - Richard W.M. Jones (rjones@redhat.com)

diff -ur nfs-utils-1.1.4/support/export/Makefile.am nfs-utils-1.1.4.buildfix/support/export/Makefile.am
--- nfs-utils-1.1.4/support/export/Makefile.am	2008-10-17 15:20:09.000000000 +0100
+++ nfs-utils-1.1.4.buildfix/support/export/Makefile.am	2009-03-02 18:46:59.000000000 +0000
@@ -40,9 +40,9 @@
 $(GENFILES_H): %.h: %.x $(RPCGEN)
 	test -f $@ && rm -rf $@ || true
 	$(RPCGEN) -h -o $@ $<
-	rm -f $(top_builddir)/support/include/mount.h
-	$(LN_S) ../export/mount.h $(top_builddir)/support/include/mount.h
+	rm -f $(top_srcdir)/support/export/mount.h
+	$(LN_S) $(abs_top_builddir)/support/export/mount.h $(top_srcdir)/support/export/mount.h
 
 MAINTAINERCLEANFILES = Makefile.in
 
-CLEANFILES = $(GENFILES) $(top_builddir)/support/include/mount.h
+CLEANFILES = $(GENFILES) $(top_srcdir)/support/export/mount.h
diff -ur nfs-utils-1.1.4/utils/mount/Makefile.am nfs-utils-1.1.4.buildfix/utils/mount/Makefile.am
--- nfs-utils-1.1.4/utils/mount/Makefile.am	2008-10-17 15:20:09.000000000 +0100
+++ nfs-utils-1.1.4.buildfix/utils/mount/Makefile.am	2009-03-03 08:50:30.000000000 +0000
@@ -17,6 +17,8 @@
 		    parse_opt.h parse_dev.h \
 		    nfs4_mount.h nfs_mount4.h stropts.h version.h
 
+mount_nfs_CFLAGS = -I$(top_builddir)/support/export
+
 mount_nfs_LDADD = ../../support/nfs/libnfs.a \
 		  ../../support/export/libexport.a
 
diff -ur nfs-utils-1.1.4/utils/statd/Makefile.am nfs-utils-1.1.4.buildfix/utils/statd/Makefile.am
--- nfs-utils-1.1.4/utils/statd/Makefile.am	2008-10-17 15:20:09.000000000 +0100
+++ nfs-utils-1.1.4.buildfix/utils/statd/Makefile.am	2009-03-02 18:49:10.000000000 +0000
@@ -51,10 +51,12 @@
 $(GENFILES_H): %.h: %.x $(RPCGEN)
 	test -f $@ && rm -rf $@ || true
 	$(RPCGEN) -h -o $@ $<
+	rm -f $(top_srcdir)/utils/statd/sm_inter.h
+	$(LN_S) $(abs_top_builddir)/utils/statd/sm_inter.h $(top_srcdir)/utils/statd/sm_inter.h
 
 MAINTAINERCLEANFILES = Makefile.in
 
-CLEANFILES = $(GENFILES)
+CLEANFILES = $(GENFILES) $(top_srcdir)/utils/statd/sm_inter.h
 
 #######################################################################
 # The following allows the current practice of having
